### ISSUE TITLE
WebSocket error: Error: connect ECONNREFUSED 172.30.33.1:443

### ISSUE THREAD

**Issue by Vartkat** [andreypopov/node-red-contrib-deconz#54]
I'm running on Ubuntu 19 in Docker on a Raspberry 4.

** Describe issue **
First I didn't succeed in getting the API Key using the magic button so I used the manual method from command line :

`$ curl -d ‘{“devicetype”: “[give the application name, for instance Node-RED]”}’ -H “Content-Type: application/json” -X POST http://[your local ip address for Hassio / deCONZ]:40850/api`

As the answer came as expected I thought the IP and port were good so I copy/pasted the API key.
Tested the API key : 

`$ curl -d '{"devicetype": "Node-RED"}' -H "Content-Type: application/json" -X GET http://172.30.33.1:40850/api/<API KEY>/lights`

As I got a list of my lights I thought everything was ok but now when I click the Refresh device list in the Deconz-Input node, nothing comes.

I did a simple flow Deconz-input--->Debug

the debug log ways : 

> WebSocket error: Error: connect ECONNREFUSED 172.30.33.1:443

** Your flow here **
`[{"id":"48b2e134.deffc","type":"tab","label":"Flow 2","disabled":false,"info":""},{"id":"d9247bfc.bffb38","type":"deconz-input","z":"48b2e134.deffc","name":"","device":null,"device_name":null,"state":"","output":"always","outputAtStartup":true,"x":120,"y":140,"wires":[["8e0e5505.a0b618"],["8e0e5505.a0b618"]]},{"id":"8e0e5505.a0b618","type":"debug","z":"48b2e134.deffc","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","x":340,"y":160,"wires":[]}]
`
** Screens / Logs **

** Hardware / Software **
  - raspberry 4 
  - version of deconz gateway: [2.05.72 / 12/12/2019] 
  - version of node-red-contrib-deconz: [1.1.9]

**Comment by andreypopov** [comment]
It is not bug. check your docker settings, open ports
I dont use it, so I cant explain you.

**Comment by Vartkat** [comment]
I tried to ping to see if it was a matter of open ports or firewall but nothing stops a ping. 
I used the curl command from within Docker containers and it works 
- from Node Red
- from Deconz
- from homeassistant
- from hassio_supervisor
As I got an answer with this IP (172.30.33.1) I assume it's the right IP. 
Which path takes the communication between Home Assistant, Hassio, Node Red and node-red-contrib-deconz, who is talking to who ? Where could be the blockade ?
Was it a good idea to use the manual method ? 

Can you help me investigate.

**Comment by Vartkat** [comment]
Following some tails on the forums I managed to get the devices list :
- Server 172.30.33.1
- port 40850
- Websocket port 40860
but still, the websocket port is giving a bunch of 
WebSocket error: Error: connect ECONNREFUSED 172.30.33.1:40860...

**Comment by Vartkat** [comment]
Finally, I had to nmap 172.30.33.1 to find open ports, 8081 seems to be the right anwser.
So final config for Docker is
-Server 172.30.33.1
-port 40850
-Websocket port 8081
- SSL off (in my case)
hope this helps

**Comment by santaklon** [comment]
I'm having the same issue, however i'm running Node Red within Hassio, not Docker. Everything works fine, however debugging is impossible like this, since the debug-messages get drowned in this ever-repeating Error message! Any of you wise people have input on what i could do about it? I am quite the noob, so please dumb it down for me ;-)

**Comment by increddibelly** [comment]
I had the same issue, not wanting to map to a 172xxx address as they tend to move around. in the end,  I got it to work by using a nodered http request nod and POSTing { "devicetype": "nodered" } to http://192.168.xxx.yyy:40850/api/
that gave me a { username : API_KEY_HERE } response. 
make sure to set the web request node to return the plain utf8 string so you can read any error messages yourself. 
Usually you'll get a "link button not pressed" message (you need to go to DeCONZ ui to Authenticate an app and run the POST again in the next 60 seconds) or Closed Connection message (sudo ufw allow 40850 / sudo ufw disable / sudo ufw enable)

it feels a bit flaky to get it started, but since it's been up it's been working flawlessly.

**Comment by santaklon** [comment]
Thanks for the reply. Quite honestly i have no idea what 90% of the word mean that you used, but I'll try to find out!

**Comment by increddibelly** [comment]
does this help?
```
[
    {
        "id": "6cc6042d.716ccc",
        "type": "http request",
        "z": "a4703b12.6b3078",
        "name": "get api key from PhosconGW",
        "method": "POST",
        "ret": "txt",
        "paytoqs": false,
        "url": "http://your_ip_here:40850/api/",
        "tls": "",
        "persist": true,
        "proxy": "",
        "authType": "",
        "x": 540,
        "y": 200,
        "wires": [
            [
                "760c23b.752bedc"
            ]
        ]
    },
    {
        "id": "76988d7c.e71374",
        "type": "inject",
        "z": "a4703b12.6b3078",
        "name": "Get DeCONZ API key",
        "topic": "",
        "payload": "{\"devicetype\":\"nodered\"}",
        "payloadType": "json",
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "x": 240,
        "y": 240,
        "wires": [
            [
                "6cc6042d.716ccc",
                "760c23b.752bedc"
            ]
        ]
    },
    {
        "id": "760c23b.752bedc",
        "type": "debug",
        "z": "a4703b12.6b3078",
        "name": "",
        "active": false,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "false",
        "x": 910,
        "y": 240,
        "wires": []
    }
]


```

**Comment by santaklon** [comment]
Ahm well, i guess all this goes into the http request node somehow? I'll have to look at this when i'm back home from work -never used a request node before, so...

And this might be super stupid: But i guess that request node will just output a string but does not actually change anything? How does that actually help with with my error? Or do i have to do something with that string after requesting it? Sorry i really don't get the concept of how my WebsocketError could be solved by making a http request quite yet - but then again i really am a total noob, so...

**Comment by increddibelly** [comment]
The http request will spit out a username, which is the API key you need to enter in the deconz server config node :)
